packages/deployments/contracts/contracts/messaging/libraries/RateLimited.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.15;
+
+/**
+ * @notice An abstract contract intended to manage the rate limiting aspect of spoke
+ * connector messaging. Rate limiting the number of messages we can send over a span of
+ * blocks is used to mitigate key DoSing vectors for transporting messages between chains.
+ */
+abstract contract RateLimited {
+  // ========== Custom Errors ===========
+
+  error RateLimited__rateLimited_messageSendRateExceeded();
+
+  // ============ Events ============
+
+  event SendRateLimitUpdated(address updater, uint256 newRateLimit);
+
+  // ============ Public Storage ============
+
+  /**
+   * @notice The number of blocks required between message sending events.
+   * @dev NOTE: This value is 0 by default, meaning that rate limiting functionality
+   * will naturally be disabled by default.
+   */
+  uint256 public rateLimitBlocks;
+
+  /**
+   * @notice Tracks the last block that we sent a message.
+   */
+  uint256 public lastSentBlock;
+
+  // ============ Modifiers ============
+
+  /**
+   * @notice Checks to see if we can send this block, given the current rate limit
+   * setting and the last block we sent a message. If rate limit has been surpassed,
+   * we update the `lastSentBlock` to be the current block.
+   */
+  modifier rateLimited() {
+    // Check to make sure we have surpassed the number of rate limit blocks.
+    if (lastSentBlock + rateLimitBlocks > block.number) {
```

Comment:
it looks wrong to me. shouldn't it be like the following?!?!
`if (lastSentBlock + rateLimitBlocks < block.number)`

Reply:
Actually I think @jakekidd is correct. If the last sent block plus the rate limit blocks is larger than the current block it means not enough blocks have passed, so you want to revert.